solutions.rb

```diff
@@ -0,0 +1,46 @@
+class Animal
+	@@animalCounter = 0
+
+	def initialize(kind)
+		@kind = kind
+		@@animalCounter += 1
+	end
+
+	def eat(food)
+		puts "#{@kind} is eating #{food}."
+	end
+
+	def sleep
+		@state = "asleep"
+	end
+
+	def wake
+		@state = "awake"
+	end
+
+	def self.count
+    	@@animalCounter
+  	end
+
+end
+
+class Person < Animal
+
+	def initialize(age, gender, name)
+		@kind = "person"
```

Comment:
Instead, you could use `super` to setup this initialization which is shared between the super and sub classes.